IonDotnet/Internals/Tree/SystemTreeReader.cs

```diff
@@ -0,0 +1,321 @@
+﻿using IonDotnet.Tree;
+using IonDotnet.Tree.Impl;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Text;
+
+namespace IonDotnet.Internals.Tree
+{
+    internal class SystemTreeReader : IIonReader
+    {
+        protected readonly ISymbolTable _systemSymbols;
+        protected IEnumerator<IIonValue> _iter;
+        protected IIonValue _parent;
+        protected IIonValue _next;
+        protected IIonValue _current;
+        protected bool _eof;
+        protected int _top;
+        private Object[] _stack = new Object[10];
+
+        protected SystemTreeReader(IIonValue value)
+        {
+            _systemSymbols = SharedSymbolTable.GetSystem(1);
+            _current = null;
+            _eof = false;
+            _top = 0;
+            if (value.Type() == IonType.Datagram)
+            {
+                _parent = value;
+                _next = null;
+                _iter = value.GetEnumerator();
```

Comment:
We should verify that this enumerator returns system values (i.e. symbol tables). In ion-java, it's possible to iterate over only the user values, or over all values including system values.

Reply:
Created issue #50 to allow enumeration of system values.